### PR TITLE
PMM-12463 Add telemetry envvar datasource

### PR DESCRIPTION
**Pull request by ademidoff**
PMM-12463

Link to the Feature Build: SUBMODULES-3412

- Links to related pull requests (optional).

### CODE REVIEW

File: managed/cmd/pmm-managed/main.go [hunk 1]
```diff
@@ -687,9 +687,9 @@ func main() { //nolint:cyclop,maintidx
 	victoriaMetricsConfigF := kingpin.Flag("victoriametrics-config", "VictoriaMetrics scrape configuration file path").
 		Default("/etc/victoriametrics-promscrape.yml").String()
 
-	grafanaAddrF := kingpin.Flag("grafana-addr", "Grafana HTTP API address").Default("127.0.0.1:3000").String()
-	qanAPIAddrF := kingpin.Flag("qan-api-addr", "QAN API gRPC API address").Default("127.0.0.1:9911").String()
-	dbaasControllerAPIAddrF := kingpin.Flag("dbaas-controller-api-addr", "DBaaS Controller gRPC API address").Default("127.0.0.1:20201").String()
+	grafanaAddrF := kingpin.Flag("grafana-addr", "Grafana API HTTP address").Default("127.0.0.1:3000").String()
```
[the remark this answers is not in the document]
Comment (ademidoff) [reply managed/cmd/pmm-managed/main.go:690]: It lacks consistency either way, for example: `Grafana HTTP API address` vs `QAN API gRPC API address`. And the latter is not even readable IMO :)

[the remark this answers is not in the document]
Comment (BupycHuk) [reply managed/cmd/pmm-managed/main.go:690]: I think it should be qan-api gRPC API

File: managed/services/telemetry/datasource_envvars.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			envVars = append(envVars, v)
+		}
+	}
+
+	for _, envVar := range envVars {
+		if value, ok := os.LookupEnv(envVar); ok && value != "" {
+			metrics = append(metrics, &pmmv1.ServerMetric_Metric{
+				Key:   envVar,
+				Value: value,
+			})
```
[the remark this answers is not in the document]
Comment (ademidoff) [reply managed/services/telemetry/datasource_envvars.go]: Sure, it's done.

File: managed/services/telemetry/transform.go [hunk 3]
```diff
@@ -86,6 +86,27 @@ func transformToJSON(config *Config, metrics []*pmmv1.ServerMetric_Metric) ([]*p
 	}, nil
 }
 
+func transformStripValues(config *Config, metrics []*pmmv1.ServerMetric_Metric) ([]*pmmv1.ServerMetric_Metric, error) {
```
Comment (BupycHuk) [inline managed/services/telemetry/transform.go:89]: I think we need a better naming here, maybe `transformValuesPresented` or something else

File: managed/services/telemetry/transform_test.go [hunk 4]
```diff
@@ -107,48 +107,171 @@ func Test_transformToJSON(t *testing.T) {
...
+func configEnvVars() *Config {
+	return &Config{
+		Source: "ENV_VARS",
+		Transform: &ConfigTransform{
+			Type: StripValuesTransform,
+		},
+		Data: []ConfigData{
+			{MetricName: "ENABLE_A", Column: "column-a"},
+			{MetricName: "DISABLE_B", Column: "column-b"},
+		},
```
Comment (BupycHuk) [inline managed/services/telemetry/transform_test.go:271]: this data field confuses in context of this test, seems like it's not used and can be removed

**Review by BupycHuk — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

File: managed/services/telemetry/transform.go (hunk 3 above)
Comment (ademidoff) [reply managed/services/telemetry/transform.go]: I renamed it to `transformExportValues` :)

File: managed/services/telemetry/transform_test.go (hunk 4 above)
Comment (ademidoff) [reply managed/services/telemetry/transform_test.go]: Agree... I had to yank one useless test that was relying on the `data` field )

**Pull request merged**